src/impl_1d.rs

```diff
@@ -23,5 +26,77 @@ impl<A, S> ArrayBase<S, Ix1>
             ::iterators::to_vec(self.iter().map(|x| x.clone()))
         }
     }
+
+    /// Return the element that would occupy the `i`-th position if the array
+    /// were sorted in increasing order.
+    ///
+    /// The array is shuffled **in place** to retrieve the desired element:
+    /// no copy of the array is allocated.
+    /// No assumptions should be made on the ordering of elements
+    /// after this computation.
+    ///
+    /// Complexity ([quickselect](https://en.wikipedia.org/wiki/Quickselect)):
+    /// - average case: O(`n`);
+    /// - worst case: O(`n`^2);
+    /// where n is the number of elements in the array.
+    ///
+    /// **Panics** if `i` is greater than or equal to `n`.
+    pub fn ith_mut(&mut self, i: usize) -> A
+        where A: Ord + Clone,
+              S: DataMut,
+    {
+        let n = self.len();
+        if n == 1 {
+            (&self[0]).clone()
```

Comment:
I don't have a compiler in front of me so I may be wrong, but I don't understand why you can't use `self[0].clone()`? Why do you need to take the reference before?

Idem for 2 other clones below.

Reply:
No, it is not necessary. Fixed it in every occurrence.